### PR TITLE
Mm 28

### PR DESCRIPTION
**Pull request by AnastasiyaPoleshuk**
ticket: https://magicmark.atlassian.net/jira/software/projects/MM/boards/1?selectedIssue=MM-28

### CODE REVIEW

File: getMarks/getMarks.js [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
+  const { rows: subjectsDb } = await db.query('SELECT * FROM "subjects"');
+
+  if (SubjectId > 0 && SubjectId <= subjectsDb.length) {
+    const marks = await getUserMarks(userId, SubjectId);
 
     response.marksData = {
-      SubjectId: id,
-      SubjectName: constants.CONSTANTS.MOCK_SUBJECTS[id - 1].SubjectName,
+      SubjectId: SubjectId,
+      SubjectName: subjectsDb[SubjectId - 1].name,
```
Comment (AyvengoBY) [inline getMarks/getMarks.js]: How dose this code will work if SubjectIDs will not ordered in the DB ? Example : 
1 Математика
4 Физика
2  ИЗО

File: queries/queries.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const config = path.join(__dirname, '../utils/ca.crt');
+const pool = new Pool({
+  connectionString: 'postgres://ikeuveifiydacx:cba5ff25e0483ec9cb63f9e4d0f0bda66f5ca6a83b201c08dba04074042850b9@ec2-176-34-215-248.eu-west-1.compute.amazonaws.com:5432/d10dt4qftsl1as',
+  ssl: {
+    rejectUnauthorized: false,
+    ca: fs.readFileSync(config).toString(),
+  }
+});
+
+const transaction = async (userId, subjectId, marks) => {
```
Comment (AyvengoBY) [inline queries/queries.js]: This method updates marks in the DB. Rename,  please.

File: updateMarks/updateMarks.js [hunk 3]
```diff
@@ -1,44 +1,60 @@
...
+      subjectId,
+      marks
+    );
+
+    if (transactionResponse) {
+      const dbMarks = await getUserMarks(userId, subjectId);
+
+      response.marksData = {
+        SubjectId: subjectId,
+        SubjectName: subjectsDb[subjectId - 1].name,
```
Comment (AyvengoBY) [inline updateMarks/updateMarks.js]: See comment for GetUserMarks endpoint.

File: user/user.js [hunk 4]
```diff
@@ -1,47 +1,79 @@
...
+    subjectsArr = [...subjectsArr, subject];
+  }
+
+  return subjectsArr;
+}
+
+async function createSubjectObject(userId, subjectItem) {
+  const marks = await getUserMarks(userId, subjectItem.id);
+  let subject = {
+    SubjectId: +subjectItem.id,
```
Comment (AyvengoBY) [inline user/user.js]: Why do we need +subjectItem.id ?

Comment (AnastasiyaPoleshuk) [reply user/user.js]: because type of subject id must be number in UI but DB returns string

File: queries/queries.js (hunk 2 above)
Comment (AnastasiyaPoleshuk) [reply queries/queries.js]: changed

File: getMarks/getMarks.js (hunk 1 above)
Comment (AnastasiyaPoleshuk) [reply getMarks/getMarks.js]: OK, changed

File: updateMarks/updateMarks.js (hunk 3 above)
Comment (AnastasiyaPoleshuk) [reply updateMarks/updateMarks.js]: changed

**Review by AyvengoBY — APPROVED** [review]

**Pull request merged**